src/v/kafka/server/handlers/produce.cc

```diff
@@ -330,10 +331,8 @@ static ss::future<produce_response::partition> produce_topic_partition(
       model::topic_namespace_view(model::kafka_namespace, topic.name));
 
     if (timestamp_type == model::timestamp_type::append_time) {
-        auto now = std::chrono::duration_cast<std::chrono::milliseconds>(
-          ss::lowres_clock::now().time_since_epoch());
```

Comment:
@mmaslankaprv I think you can use `lowres_system_clock` which is also fast to sample, but corresponds to system time.

Reply:
@dotnwat i think that should be part of a separate change and pushed to `timestamp::now()` so it's consistent everywhere.

Reply:
ah yeh I just noticed that timestamp is using chrono system clock, so it has the overhead.